packages/next/src/generators/application/lib/add-project.ts

```diff
@@ -63,6 +63,14 @@ export function addProject(host: Tree, options: NormalizedSchema) {
         buildTarget: `${options.projectName}:build:production`,
       },
     };
+  } else {
+    // We need to add the serve-static target so that the port can be configured.
+    targets['serve-static'] = {
```

Comment:
it goes to this condition when hasPlugin is true, don't you have the getStaticServeTargetConfig to set the serve-static target?

Reply:
I think we should skip this when plugin is added. Mixing inferred and explicit targets is something users can do themselves, but we should not generate it that way.